translate-c std.meta.cast can't cast to *c_void

**Zig version: 0.8.0-dev.1431+c760532be**

**Repro:**
```zig
    const T = struct { x: i32 };
    var a = std.meta.cast(?*T, std.meta.cast(?*c_void, &T{ .x = 10 }));
    std.log.info("x={}", .{ a.?.*.x });
```

**Error**
```
$ zig build -Dtarget=native-native-gnu
C:\Dev\Tools\zig-0.8.0-dev.1431+c760532be\lib\zig\std\meta.zig:910:17: error: no align available for type 'c_void'
            if (@typeInfo(dest_opt.child) == .Pointer) {
                ^
.\src\main.zig:33:45: note: called from here
    var a = std.meta.cast(?*T, std.meta.cast(?*c_void, &T{ .x = 10 }));
                                            ^
.\src\main.zig:16:20: note: called from here
pub fn main() void {
                   ^
C:\Dev\Tools\zig-0.8.0-dev.1431+c760532be\lib\zig\std\meta.zig:917:70: note: referenced here
                        return @ptrCast(DestType, @alignCast(dest_ptr.alignment, target));
                                                                     ^
C:\Dev\Tools\zig-0.8.0-dev.1431+c760532be\lib\zig\std\meta.zig:917:51: note: referenced here
                        return @ptrCast(DestType, @alignCast(dest_ptr.alignment, target));
                                                  ^
```

Error found translating [nuklear library](https://github.com/Immediate-Mode-UI/Nuklear/blob/master/nuklear.h) macro:
```c
#define nk_ptr_add(t, p, i) ((t*)((void*)((nk_byte*)(p) + (i))))
```

```zig
pub fn nk_ptr_add(t: anytype, p: anytype, i: anytype) callconv(.Inline) [*c]t {
    return @import("std").meta.cast([*c]t, @import("std").meta.cast(?*c_void, @import("std").meta.cast([*c]nk_byte, p) + i));
}
```

--- a/lib/std/meta.zig
+++ b/lib/std/meta.zig
@@ -970,12 +970,15 @@ fn castPtr(comptime DestType: type, target: anytype) DestType {
 
     if (source.is_const and !dest.is_const or source.is_volatile and !dest.is_volatile)
         return @intToPtr(DestType, @ptrToInt(target))
+    else if (@typeInfo(dest.child) == .Opaque)
+        // dest.alignment would error out
+        return @ptrCast(DestType, target)
     else
         return @ptrCast(DestType, @alignCast(dest.alignment, target));
 }
 
 fn ptrInfo(comptime PtrType: type) TypeInfo.Pointer {
-    return switch(@typeInfo(PtrType)){
+    return switch (@typeInfo(PtrType)) {
         .Optional => |opt_info| @typeInfo(opt_info.child).Pointer,
         .Pointer => |ptr_info| ptr_info,
         else => unreachable,
@@ -1010,6 +1013,8 @@ test "std.meta.cast" {
 
     testing.expectEqual(@intToPtr(*u8, 2), cast(*u8, @intToPtr(*const u8, 2)));
     testing.expectEqual(@intToPtr(*u8, 2), cast(*u8, @intToPtr(*volatile u8, 2)));
+
+    testing.expectEqual(@intToPtr(?*c_void, 2), cast(?*c_void, @intToPtr(*u8, 2)));
 }
 
 /// Given a value returns its size as C's sizeof operator would.

